src/store/actions/userSession.actions.js

```diff
@@ -1,11 +1,73 @@
+import axios from 'axios';
 import history from '../configuration/history';
-import client, { getAccessToken, setAccessToken } from '../api';
+import { getAccessToken, setAccessToken } from '../api';
 import {
     CHECK_LOGGED_STATUS,
     LOGIN_START,
     LOGIN_SUCCESS,
     LOGIN_FAILURE,
 } from '../constants';
+import { requestStart, requestComplete } from './request.actions';
+import { getRandomString } from '../../helpers/getRandomString';
+
+const clientId = process.env.REACT_APP_CLIENT_ID;
+const scope =  process.env.REACT_APP_SCOPE;
+const state = getRandomString();
+const URL_EXCHANGE_CODE_ON_TOKEN = process.env.REACT_APP_EXCHANGE_CODE_ON_TOKEN_URL;
+const URL_GET_ACCESS_TOKEN = process.env.REACT_APP_ACCESS_TOKEN_URL;
+const URL_AUTHORIZE = process.env.REACT_APP_AUTHORIZE_URL;
+
+export const getJWTtoken = (hash) => async dispatch => {
+    const getParameterByName = (name, url) => {
+        if (!url) url = window.location.href;
+        name = name.replace(/[\[\]]/g, '\\$&');
+        const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)');
+        const results = regex.exec(url);
+
+        if (!results) return null;
+        if (!results[2]) return '';
```

Comment:
Is the linter happy with this? :D

Reply:
with this totally fine 😄  But it's complaining about another line with regexp 😆